pkg/kubelet/kubelet.go

```diff
@@ -2044,6 +2123,15 @@ func (kl *Kubelet) ResyncInterval() time.Duration {
 
 // ListenAndServe runs the kubelet HTTP server.
 func (kl *Kubelet) ListenAndServe(address net.IP, port uint, tlsOptions *server.TLSOptions, auth server.AuthInterface, enableDebuggingHandlers, enableContentionProfiling bool) {
+	if utilfeature.DefaultFeatureGate.Enabled(features.RotateKubeletServerCertificate) && tlsOptions != nil {
+		tlsOptions.Config.GetCertificate = func(*tls.ClientHelloInfo) (*tls.Certificate, error) {
```

Comment:
I'd expect this to be done by the caller, not mutating tlsOptions inside ListenAndServe